src/com/flashfla/net/Multiplayer.as

```diff
@@ -663,19 +695,44 @@ package com.flashfla.net
 
                 vars["arc_engine" + currentUser.id] = null;
                 vars["arc_replay" + currentUser.id] = null;
+
+                // If no opponents, set the room's level to the currentUser's level
+                // A player spectates and there is a player left.
+                // A player spectates and there is no players left.
+                var remainingPlayer:User = null;
+                for each (var player:User in room.players)
+                {
+                    if (player.playerIdx != currentUserIdx)
+                    {
+                        remainingPlayer = player;
+                    }
+                }
+
+                if (remainingPlayer != null)
+                {
+                    vars["GAME_LEVEL"] = remainingPlayer.userLevel;
+                }
+                else
+                {
+                    vars["GAME_LEVEL"] = -1;
+                }
             }
 
+
+
             // Send room vars to server
             sendRoomVariables(room, vars);
         }
 
         /**
          * Sends the current user's "player" variables (if any) to the server.
          */
-        private function sendCurrentUserRoomVariables(room:Room):void
+        private function sendCurrentUserRoomVariables(room:Room, joining:Boolean = true, handlingLeaver:Boolean = false):void
```

Comment:
Would be nice if you could add the parameter descriptions in this function's docs. It's not entirely clear what they imply, especially `handlingLeaver`.